### PR TITLE
SUIT-13329 Add typing information as separate package

### PR DESCRIPTION
**Pull request by andriichumak**

### CODE REVIEW

File: packages/types/lib/condition.d.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	}>,
+};
+
+export type ElementCondition = ElementExistsCondition
+	| ElementDoesNotExistCondition
+	| ElementVisibleCondition
+	| ElementMatchesJavaScriptCondition
+	| ElementPropertiesCondition;
+
+export type JavaScriptExpressionCondition = {
```
Comment (AntonSBilan) [inline packages/types/lib/condition.d.ts]: what about bright script?

Comment (andriichumak) [reply packages/types/lib/condition.d.ts]: We do not support BrightScript as part of test lines at the moment (it was commented out due to stability issues). We can re-add it at any point

File: packages/types/lib/comparator.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export type StringComparator = '=' | '!=' | '~' | '!~' | '^' | '!^' | '$' | '!$';
+
+export type NumberComparator = '=' | '!=' | '+-' | '>' | '>=' | '<' | '<=';
+
+export type ColorComparator = '=' | '!=' | '+-';
+
+export type JavaScriptComparator = 'matches';
```
Comment (JeKachan) [inline packages/types/lib/comparator.d.ts]: For what kind of properties used this comparator type?

File: packages/types/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+    "preset": "ts-jest",
+    "transform": {
+        "^.+\\.(j|t)s?$": "ts-jest"
+    },
+    "testMatch": [
+        "<rootDir>/lib/__tests__/*.test.ts"
+    ],
+    "collectCoverageFrom": [
+        "<rootDir>/lib/index.ts"
```
Comment (JeKachan) [inline packages/types/jest.config.js]: tests will be added later?

File: packages/types/package.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "@suitest/types",
+  "version": "2.0.4",
```
Comment (JeKachan) [inline packages/types/package.json]: probably mistake?

File: packages/types/lib/comparator.d.ts (hunk 2 above)
Comment (andriichumak) [reply packages/types/lib/comparator.d.ts]: This is used in e.g. `Assert [something] matches [js]`. This is not specifically for properties, but rather a comparator in general - both props and a complete conditions.

File: packages/types/jest.config.js (hunk 3 above)
Comment (andriichumak) [reply packages/types/jest.config.js]: No, there is no point to test types. I'll drop this.

File: packages/types/package.json (hunk 4 above)
Comment (andriichumak) [reply packages/types/package.json]: We are using single version for all packages in this repo. So, it's set to same value as in root package json. Will be updated when we are ready for release.

**Pull request merged**